File: backend/backend/src/main/java/com/inspectionapp/backend/dto/response/JwtResponse.java
package com.inspectionapp.backend.dto.response;

/**
 * Returned by {@code POST /api/v1/auth/login} on successful authentication.
 *
 * <p>Clients store the {@code token} and include it in every subsequent request
 * as an HTTP header:
 * <pre>
 *   Authorization: Bearer &lt;token&gt;
 * </pre>
 *
 * @param token    the signed JWT (compact serialisation)
 * @param type     always {@code "Bearer"} — the token type expected by the
 *                 {@code Authorization} header scheme
 * @param username the authenticated user's login handle
 * @param role     the user's role ({@code ADMIN} or {@code INSPECTOR}),
 *                 stripped of the {@code ROLE_} prefix
 */
public record JwtResponse(
        String token,
        String type,
        String username,
        String role
) {
}

File: mobile/mobile/lib/features/auth/models/jwt_response.dart
/// Mirrors the backend [JwtResponse] record returned by POST /api/v1/auth/login.
class JwtResponse {
  const JwtResponse({
    required this.token,
    required this.type,
    required this.username,
    required this.role,
  });

  final String token;    // Signed JWT (compact serialisation)
  final String type;     // Always "Bearer"
  final String username;
  final String role;     // "ADMIN" or "INSPECTOR" (ROLE_ prefix stripped by backend)

  factory JwtResponse.fromJson(Map<String, dynamic> json) => JwtResponse(
    token:    json['token']    as String,
    type:     json['type']     as String,
    username: json['username'] as String,
    role:     json['role']     as String,
  );
}
